src/controllers/colormapper.cpp

```diff
@@ -0,0 +1,59 @@
+#include "controllers/colormapper.h"
+
+#include <cmath>
+#include <cstdint>
+
+#include "util/debug.h"
+
+namespace {
+double colorDistance(QRgb a, QRgb b) {
+    // This algorithm calculates the distance between two colors. In
+    // contrast to the L2 norm, this also tries take the human perception
+    // of colors into account. More accurate algorithms like the CIELAB2000
+    // Delta-E rely on sophisticated color space conversions and need a lot
+    // of costly computations. In contrast, this is a low-cost
+    // approximation and should be sufficently accurate.
+    // More details: https://www.compuphase.com/cmetric.htm
```

Comment:
observation: I've found that doing distance through HSV looks very nice compared to RGB.

Reply:
HSV would require color space conversions, and doing color distance in HSV does not take human color perception. This is why CIELAB ΔE* ("Delta-E") exists, which calculates the difference in the dedicated CIELAB color space. However, this is very costly. The algorithm we are using here is a low-cost approximation in the RGB space which is very fast and accurate enough IMHO.